src/services/components/guestBook.jsx

```diff
@@ -0,0 +1,73 @@
+//imports
+import React, { useState } from 'react';
+import { useEntries } from '../../context/EntryContext';
+import { useUser } from '../../context/UserContext';
+import './guestBook.css';
+
+export default function GuestBook() {
+  const [name, setName] = useState('');
+  const [messageEntry, setMessageEntry] = useState('');
+  const { user, setUser } = useUser();
+  const { message, setMessage } = useEntries();
+
+  function updateMessageList() {
+    if (!messageEntry) return;
+    setUser(name);
+    setMessage([...message, { name, message: messageEntry }]);
```

Comment:
you might want to use a callback and spread in prevState here instead of spreading in message, so that you can be sure it stays updated